eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -405,7 +437,26 @@ object MultiPartHandler {
                       context.log.info("rejecting blinded htlc #{} from channel {}: invoice not found", packet.add.id, packet.add.channelId)
                       replyTo ! RejectPacket(packet.add, IncorrectOrUnknownPaymentDetails(payload.totalAmount, nodeParams.currentBlockHeight))
                   }
+                  Behaviors.stopped
+                case payload: FinalPayload.Blinded =>
+                  offerManager ! OfferManager.Payment(context.self, packet.add.paymentHash, payload)
+                  waitForPayment(nodeParams, replyTo, packet.add, payload)
               }
+          }
+        }
+      }
+    }
+
+    def waitForPayment(nodeParams: NodeParams, replyTo: ActorRef, add: UpdateAddHtlc, payload: FinalPayload.Blinded): Behavior[Command] = {
+      Behaviors.setup { context =>
+        Behaviors.withMdc(Logs.mdc(category_opt = Some(LogCategory.PAYMENT), paymentHash_opt = Some(add.paymentHash))) {
```

Comment:
No need to call `setup` / `withMdc` again, you should pass the `context` as argument.

Reply:
Done